### ISSUE TITLE
BakedLightMap causes unbaked meshes to disappear in 3.2.4 RC 1

### ISSUE THREAD

**Issue by ellenhp** [godotengine/godot#45584]
**Godot version:**
3.2.4 RC 1

**OS/device including version:**
Debian Bullseye with open-source AMD drivers and the GLES2 backend. I'm running an RX570 but I don't think that'll end up mattering.

**Issue description:**
Adding a lightmap to the scene and baking it causes non-lightmapped meshes to disappear. I'm not super familiar with 3d game dev but my understanding is they should be lit dynamically instead of disappearing. The symptoms seem identical to #24550

**Steps to reproduce:**
Create a new scene and add a light and a camera. Import a mesh, and perform the UV2 unwrap. Add the mesh to two different MeshInstances, and mark only one of them as "use in baked light". Add a BakedLightMap node and bake its lightmap. Export the project with the official HTML5 template for 3.2.4 RC1, host it somewhere and observe the results in both firefox and chromium.

![export_firefox](https://user-images.githubusercontent.com/1936763/106361593-9dccd380-62d3-11eb-90e9-4652fb65f79e.png)
![export_chromium](https://user-images.githubusercontent.com/1936763/106361598-a4f3e180-62d3-11eb-9923-d6deb634f394.png)

It works fine in the editor though
![editor](https://user-images.githubusercontent.com/1936763/106361631-cfde3580-62d3-11eb-94fb-7325d804642f.png)

**Minimal reproduction project:**
[lightmap_bug_repro.zip](https://github.com/godotengine/godot/files/5898272/lightmap_bug_repro.zip)

**Additional Info**

The logs point to an issue with shader compilation. I managed to track this one down. It appears that the logic deciding whether or not to enable the lightmap baking code (with the cone tracing and all that) doesn't look at whether or not it's in the editor, so it compiles the cone tracing into the shader even in the HTML5 export. This ends up being an issue because it seems like there might be another array-related issue with the lightmapping code. Either that or the old one resurfaced.

```
**ERROR**: SceneShaderGLES2: Fragment shader compilation failed: index.js:9:17349
ERROR: 0:721: 'first-class array' : not supported index.js:9:17349
<empty string> index.js:9:17349
   At: drivers/gles2/shader_gles2.cpp:130:_display_error_with_code() - SceneShaderGLES2: Fragment shader compilation failed: index.js:9:17349
ERROR: 0:721: 'first-class array' : not supported index.js:9:
```

And here's one potential fix that seems to work on my hardware. It makes sense to me. I don't ever want to _capture_ lightmap data unless I'm in the editor.

```
diff --git a/drivers/gles2/rasterizer_scene_gles2.cpp b/drivers/gles2/rasterizer_scene_gles2.cpp
index ff27f11333..a67e7367a9 100644
--- a/drivers/gles2/rasterizer_scene_gles2.cpp
+++ b/drivers/gles2/rasterizer_scene_gles2.cpp
@@ -2400,7 +2400,7 @@ void RasterizerSceneGLES2::_render_render_list(RenderList::Element **p_elements,
                                rebind_reflection = true;
                        }
 
-                       use_lightmap_capture = !unshaded && !accum_pass && !e->instance->lightmap_capture_data.empty();
+                       use_lightmap_capture = !unshaded && !accum_pass && !e->instance->lightmap_capture_data.empty() && Engine::get_singleton()->is_editor_hint();
 
                        if (use_lightmap_capture != prev_use_lightmap_capture) {
 
```

**Comment by Calinou** [comment]
Related to #45574.

**Comment by ellenhp** [comment]
> If you apply the change you've described, this means dynamic objects will no longer be correctly lit in an exported project.

Interesting. It definitely is still lighting dynamic objects. I compiled a custom HTML5 export template with the patch I provided to export a game of mine to itch.io yesterday. The exported game runs on both my chromebook and my desktop, along with a number of friends computers. Now I'm not sure why it works...

**Comment by ellenhp** [comment]
It's possible I completely disabled lightmapping outside of the editor, looking at that patch. I'm not sure. I'll wait for an upstream fix though, since this area of the engine is completely foreign to me.

**Comment by WeaselShoppe** [comment]
I had similar issue. Fixed with this PR:
https://github.com/godotengine/godot/pull/45622

**Comment by WeaselShoppe** [comment]
Some before and after images:
<img width="1909" alt="baked_light_map_error_wgl1 0" src="https://user-images.githubusercontent.com/71855294/106409856-239d6b80-6407-11eb-9aab-3cf1f7ddae04.PNG">
<img width="1908" alt="baked_light_map_no_error_wgl 1 0" src="https://user-images.githubusercontent.com/71855294/106409858-239d6b80-6407-11eb-9e9f-18fc61cddec3.PNG">

**Comment by akien-mga** [comment]
Fixed by #45622.
